internal/pkg/model/relation.go

```diff
@@ -0,0 +1,91 @@
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+const (
+	VariablesForRelType = RelationType(`variablesFor`)
+)
+
+type RelationType string
+
+func (t RelationType) String() string {
+	return string(t)
+}
+
+func (t RelationType) Type() RelationType {
+	return t
+}
+
+func newEmptyRelation(t RelationType) (Relation, error) {
+	switch t {
+	case VariablesForRelType:
+		return &VariablesForRelation{RelationType: VariablesForRelType}, nil
+	default:
+		return nil, fmt.Errorf(`unexpected RelationType "%s"`, t)
+	}
+}
+
+// Relation between objects, eg. config <-> config.
+type Relation interface {
```

Comment:
Interface pre lubovolnu `Relation` medzi 2 objektmi.

Bude sa to pouzivat pre `Variables` (tento PR), ... a neskor aj pre `Schedules`.